client/consensus/manual-seal/src/seal_new_block.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Block sealing utilities
+
+use crate::{Error, rpc};
+use std::sync::Arc;
+use sp_runtime::{
+	traits::{Block as BlockT, Header as HeaderT},
+	generic::BlockId,
+};
+use transaction_pool::txpool;
+use rpc::CreatedBlock;
+
+use consensus_common::{
+	self, BlockImport, Environment, Proposer,
+	ForkChoiceStrategy, BlockImportParams, BlockOrigin,
+	ImportResult, SelectChain,
+	import_queue::BoxBlockImport,
+};
+use sp_blockchain::HeaderBackend;
+use client_api::backend::Backend as ClientBackend;
+use hash_db::Hasher;
+use std::collections::HashMap;
+use std::time::Duration;
+use std::marker::PhantomData;
+
+/// params for sealing a new block
+pub struct SealBlockParams<'a, B: BlockT, C, CB, E, P: txpool::ChainApi, H> {
+	/// if true, empty blocks(without extrinsics) will be created.
+	/// otherwise, will return Error::EmptyTransactionPool.
+	pub create_empty: bool,
+	/// instantly finalize this block?
+	pub finalize: bool,
+	/// specify the parent hash of the about-to-created block
+	pub parent_hash: Option<<B as BlockT>::Hash>,
+	/// sender to report errors/success to the rpc.
+	pub sender: rpc::Sender<CreatedBlock<<B as BlockT>::Hash>>,
+	/// transaction pool
+	pub pool: Arc<txpool::Pool<P>>,
+	/// client backend
+	pub back_end: Arc<CB>,
+	/// Environment trait object for creating a proposer
+	pub env: &'a mut E,
+	/// SelectChain object
+	pub select_chain: &'a C,
+	/// block import object
+	pub block_import: &'a mut BoxBlockImport<B>,
+	/// inherent data provide
+	pub inherent_data_provider: &'a inherents::InherentDataProviders,
+	/// phantom data to pin the hasher type.
+	pub _phantom: PhantomData<H>
```

Comment:
Same here, I don't think the `H` type is useful anywhere and it doesn;'t need to be bound to the params.